src/ray/pubsub/subscriber.cc

```diff
@@ -76,13 +76,14 @@ bool SubscriberChannel<KeyIdType>::CheckNoLeaks() const {
 }
 
 template <typename KeyIdType>
-void SubscriberChannel<KeyIdType>::HandlePublishedMessage(
-    const rpc::Address &publisher_address, const rpc::PubMessage &pub_message) {
+std::shared_ptr<SubscriptionCallback>
```

Comment:
No need for std::shared_ptr. SubscriptionCallback should be enough.

You can return nullptr and do checking like ptr.

Reply:
I actually have a cpp question here. Isn't the callback including captured values? (thus pretty expensive to copy)?

Reply:
Thanks this is awesome!